shanghai/plugin_system.py

```diff
@@ -76,42 +81,59 @@ def __getattr__(self, item):
             return self.plugin_registry[item].module
         raise AttributeError(item)
 
-    def load_plugin(self, identifier, *, dependency_path=None, is_core=False):
+    def load_all_plugins(self):
+        plugins_to_load = {}
+
+        for search_path in self.plugin_search_paths:
+            for module_path in search_path.glob("*.py"):
+                identifier = module_path.stem
+                plugins_to_load.setdefault(identifier, module_path)
+
+        loaded_before = set(self.plugin_registry.keys())
+        for identifier, module_path in plugins_to_load.items():
+            if identifier in loaded_before:
+                self.logger.warn(f"Plugin {identifier!r} already exists")
+                continue
+            elif identifier in self.plugin_registry:
+                continue  # loaded as a dependency
+
+            plugin = self._load_plugin_as_module(module_path, identifier)
+            self._register_plugin(plugin)
+
+    def load_plugin(self, identifier, *, dependency_path=(), is_core=False):
         if not identifier.isidentifier():
             raise ValueError(f"Invalid plugin name. {identifier!r} contains invalid symbol(s).")
         if keyword.iskeyword(identifier):
             raise ValueError(f"Invalid plugin name. {identifier!r} is a built-in keyword.")
 
-        module_name = f'{__package__}.{self.namespace}.{identifier}'
-
-        if dependency_path is None:
-            dependency_path = []
         if identifier in self.plugin_registry:
             if not dependency_path:
-                self.logger.warn('Plugin', identifier, 'already exists.')
+                self.logger.warn(f"Plugin {identifier!r} already exists")
             return self.plugin_registry[identifier]
+
         for search_path in self.plugin_search_paths:
             try:
                 module_path = self._find_module_path(search_path, identifier)
             except OSError:
-                pass
+                continue
```

Comment:
If we use `continue` here, we can remove the `else:` next line and save one indentation. Though it's hardly important for a single break.